lib/index.js

```diff
@@ -536,12 +536,12 @@ module.exports = {
 									node: node,
 									messageId: "neverGlobal",
 								});
+								return;
 							}
-							else if (foundThis && nestedThis) {
-								let parentArrow = getParentArrowFunction(context.getAncestors());
-								if (parentArrow) {
-									thisFoundIn.add(parentArrow);
-								}
+							
+							let parentArrow = getParentArrowFunction(context.getAncestors());
+							if (foundThis && parentArrow) {
+								thisFoundIn.add(parentArrow);
```

Comment:
with this approach, the `foundThis` boolean is being checked twice in the same block, and also, the `parentArrow` is computed even if no `this` has been found to need that value. definitely minor nits, but any reason not to structure the `if..else` logic the way I sketched out in the previous review comment?

Reply:
logic is to reduce nesting, but ok)